### ISSUE TITLE
plug-in wont start

### ISSUE THREAD

**Issue by jameswestcott** [ztalbot2000/homebridge-cmd4#112]
Hi,

just updated plug-in with latest v5.0.1 from a previous v4 and the plug-in no longer runs. I get the following error message:

TypeError: Cannot read property 'filter' of undefined
    at Cmd4Accessory.setupAccessoryFakeGatoService (C:\Users\james\AppData\Roaming\npm\node_modules\homebridge-cmd4\Cmd4Accessory.js:1165:64)
    at Cmd4Accessory.createServicesForStandaloneAccessoryAndItsChildren (C:\Users\james\AppData\Roaming\npm\node_modules\homebridge-cmd4\Cmd4Accessory.js:556:17)
    at new Cmd4Accessory (C:\Users\james\AppData\Roaming\npm\node_modules\homebridge-cmd4\Cmd4Accessory.js:300:15)
    at C:\Users\james\AppData\Roaming\npm\node_modules\homebridge\src\server.ts:350:50
    at Array.forEach (<anonymous>)
    at Server.loadAccessories (C:\Users\james\AppData\Roaming\npm\node_modules\homebridge\src\server.ts:270:29)
    at Server.start (C:\Users\james\AppData\Roaming\npm\node_modules\homebridge\src\server.ts:162:12)

There are multiple entries in accessories in config.json like this one:
    "accessories": [
        {
            "accessory": "Cmd4",
            "type": "TemperatureSensor",
            "name": "Bathroom1Temperature",
            "displayName": "Bathroom1Temperature",
            "manufacturer": "cmd4",
            "model": "TemperatureSensor",
            "serialNumber": "10001",
            "CurrentTemperature": 0,
            "statusActive": "TRUE",
            "StatusFault": "NO_FAULT",
            "StatusTampered": "NOT_TAMPERED",
            "StatusLowBattery": "BATTERY_LEVEL_NORMAL",
            "polling": [
                {
                    "Characteristic": "CurrentTemperature",
                    "Interval": 60,
                    "Timeout": 10000
                }
            ],
            "fakegato": {
                "eve": "room",
                "temp": "CurrentTemperature",
                "humidity": "0",
                "ppm": "0",
                "storage": "fs"
            },
            "stateChangeResponseTime": 3,
            "state_cmd": "node C:\\Users\\james\\.homebridge\\instance2\\temperaturehumidity1.js"
        },

**Comment by ztalbot2000** [comment]
Hi,

  So nothing in Cmd4 v5 touched FakeGato.  I'll poke around.  Do you know
if FakeGato has changed or what version you have installed?

Thanks,
John Talbot

**Comment by jameswestcott** [comment]
I'm not sure if it's changed. Am now running 0.6.1. Not sure what version I had before. Will also take a look.

**Comment by ztalbot2000** [comment]
Hi,

  I found the bug. As I said I did not touch FakeGato.  There is a check to
see if the Fakegato characteristic is polled.  In your case it fails.  I
will put out a fix in the next couple of hours.

I'm also going to put in some unit tests that will find this if it were to
happen in the future.  Frankly, I thought there were already. With 9513
test cases, you think there would be. I guess not for this.

Sorry about that,
John
